src/compute/src/sink/persist_sink.rs

```diff
@@ -622,7 +624,10 @@ where
                 // NOTE: We schedule ourselves to make sure that we keep
                 // heartbeating even in cases where there are no changes in our
                 // inputs (updates or frontier changes).
-                activator.activate_after(Duration::from_secs(60));
+                if last_heartbeat_activation <= Instant::now() + Duration::from_secs(60) {
```

Comment:
```suggestion
                if last_heartbeat_activation + Duration::from_secs(60) <= Instant::now() {
```

Otherwise the condition is always true and we'd have the same behavior as before.